### DISCUSSION TITLE
工具调用是格式化成JSON添加到history上，还是直接在微调中加进去

### DISCUSSION THREAD

**Discussion by vshencc** [zai-org/ChatGLM3#560] [Q&A] [answered]
似乎两种方式都可以，我尝试了在每次对话开始前将工具调用的JSON插入到history头部，这样是不是会占用对话长度？
把工具调用在微调时就加入训练，效果会不会好一点

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
会占用，本身工具调用的json也算长度 #253 这里可以咨询微调的问题
